addons/http_routing/models/ir_http.py

```diff
@@ -337,220 +341,241 @@ def get_nearest_lang(cls, lang_code):
-        for key in ('path', 'full_path', 'url', 'base_url'):
-            request.httprequest.__dict__.pop(key, None)
-
-        return cls._dispatch()
+    def reroute(cls, path, query_string=None):
+        """
+        Rewrite the current request URL using the new path and query
+        string. This act as a light redirection, it does not return a
+        3xx responses to the browser but still change the current URL.
+        """
+        if query_string is None:
+            query_string = request.httprequest.environ['QUERY_STRING']
+
+        # Change the WSGI environment
+        environ = request.httprequest.environ.copy()
+        environ['PATH_INFO'] = path
+        environ['QUERY_STRING'] = query_string
+        environ['RAW_URI'] = f'{path}?{query_string}'
+        # REQUEST_URI left as-is so it still contains the original URI
+
+        # Create and expose a new request from the modified WSGI env
+        httprequest = werkzeug.wrappers.Request(environ)
+        httprequest.parameter_storage_class = (
+            werkzeug.datastructures.ImmutableOrderedMultiDict)
+        threading.current_thread().url = httprequest.url
+        request.httprequest = httprequest
 
     @classmethod
-    def _postprocess_args(cls, arguments, rule):
-        super(IrHttp, cls)._postprocess_args(arguments, rule)
+    def _pre_dispatch(cls, rule, args):
+        super()._pre_dispatch(rule, args)
 
-        try:
-            _, path = rule.build(arguments)
-            assert path is not None
-        except odoo.exceptions.MissingError:
-            return cls._handle_exception(werkzeug.exceptions.NotFound())
-        except Exception as e:
-            return cls._handle_exception(e)
-
-        if getattr(request, 'is_frontend_multilang', False) and request.httprequest.method in ('GET', 'HEAD'):
-            generated_path = werkzeug.urls.url_unquote_plus(path)
-            current_path = werkzeug.urls.url_unquote_plus(request.httprequest.path)
-            if generated_path != current_path:
-                if request.lang != cls._get_default_lang():
-                    path = '/' + request.lang.url_code + path
-                if request.httprequest.query_string:
-                    path += '?' + request.httprequest.query_string.decode('utf-8')
-                return request.redirect(path, code=301)
+        # Lazy init of GeoIP resolver
+        if odoo._geoip_resolver is None:
+            geofile = config.get('geoip_database')
+            try:
+                odoo._geoip_resolver = GeoIPResolver.open(geofile) or False
+            except Exception as e:
+                _logger.warning('Cannot load GeoIP: %s', ustr(e))
+
+        # Set geoip on the session
+        if 'geoip' not in request.session:
+            record = {}
+            if odoo._geoip_resolver and request.httprequest.remote_addr:
+                record = odoo._geoip_resolver.resolve(request.httprequest.remote_addr) or {}
+            request.session['geoip'] = record
```

Comment:
YAGNI, we will isolate it in an overridable function the day we need to override it ;)

Reply:
Ok, I can survive with it...
But it just strange to remove it, it was done previously. And unfortunately, no, the community/external dev cannot edit easily the base code to make their customs.

YAGNI ? please... On this same PR you add some try/except around function that cannot raise any exception in case of somebody add some codes and in case that this code raises an exception... that you call programming preventive. And now make your code modular and overridable is stupid?

Reply:
AH! If it was overridable before then for sure I shouldn't remove it, I'm bringing the method back